apps/swirl-docs/src/components/Tokens/TokenPreview.tsx

```diff
@@ -0,0 +1,104 @@
+import { Token, typographyTypes } from "@swirl/lib/tokens";
+import { FunctionComponent } from "react";
+
+export type TokenPreviewProps = {
+  token: Token;
+};
+
+const TokenPreview: FunctionComponent<TokenPreviewProps> = ({ token }) => {
+  if (token.type === "color") {
```

Comment:
note: currently the structure is inspired by the polaris solution. But to be honest i do not like the nested ilf else structure, i will think about a solution that is more "convenient" to read, so maybe the apis and structure of this component will change